components/eamxx/scripts/update-all-pip

```diff
@@ -0,0 +1,16 @@
+#! /bin/bash -xe
+
+# Our python scripts use pip to manage externals python
+# dependencies. To ensure that a machine has the necessary
+# dependencies, we have a set of ensure_$depname functions available
+# in utils.py. These functions will not reliably work unless the set
+# of existing python packages are relatively up-to-date. This script
+# will update all pip packages.
+#
+# Note: netcdf4 1.6.2 would not install cleanly on weaver so we had
+# to force a downgrade to 1.6.0.
+
+python3 -m pip install --upgrade pip --user
+for item in $(python -m pip list --outdated | grep -v Package | grep -v -- '-----' | cut -f 1 -d ' '); do
```

Comment:
This bash command is a bit cryptic, and I can't understand which item names we're trying to gather. Can you add a comment?

Reply:
Ok, after running the command on my laptop, I figured out what it does. But I'm puzzled: won't this also update netcdf4?